src/lib/dnssd/Constants.h

```diff
@@ -29,7 +29,7 @@ namespace Dnssd {
  * Matter DNS host settings
  */
 
-inline constexpr size_t kHostNameMaxLength = 16; // MAC or 802.15.4 Extended Address in hex
+inline constexpr size_t kHostNameMaxLength = 40; // The max host name length for dnssd in android
```

Comment:
I guess that's OK as a workaround, but this means that Android implementation is not spec-compliant, right? Will this eventually be fixed in Android or should the spec be updated to make Android devices discoverable by all compliant Matter devices?

Reply:
I would discuss with our android team for this issue in NSDManager, which may involve a new API surface, since android is pretty fragmented, even though we make this change happen, I think it would be hard to spread this change uniformly across. I would raise this issue with TSG and discuss whether matter the spec can be updated as well.

Reply:
I would work on this for the follow-up https://github.com/project-chip/connectedhomeip/issues/33474, thanks

Reply:
Thanks for filing the follow up!

Reply:
Let's add this as a comment - why this is a workaround as well as a link to the open issue.
Otherwise the spec seems to explicitly say that we have a different limit and this causes some slight increase in RAM usage (looks like around 100 bytes ... we probably repeat this in 4 places)

Reply:
For comment I mean both keeping the spec description and describing that android is a workaround. Otherwise it is not obvious that a chip dnssd constant is `The max host name length for dnssd in android` is correct ... why was android picked against all other implementations.